C2A-bare.lsl

```diff
@@ -0,0 +1,135 @@
+/*
+    C2A-bare.lsl
+        https://github.com/ThunderRahja/combat2armor/blob/main/C2A-bare.lsl
+    
+    This script is licensed under GNU GPL 3.0. You can read the license here:
+        https://github.com/ThunderRahja/combat2armor/blob/main/LICENSE
+*/
+
+float MAX_POINTS = 100;
+float points;
+string descPrefix = "LBA.v.[C2A LBA v0.1.0:"; // + c2aName + "]," <-- added when c2aName is generated
+
+
+TakeDamage(float amount)
+{
+    points -= amount;
+    
+    // Limit points, can't be deader than already dead. However MAX_POINTS logic here could be tweaked to allow overhealth repairs, e.g. (points > MAX_POINTS * 1.25) points = MAX_POINTS * 1.25; // +25% overhealth
+    if (points < 0) points = 0;
+    else if (points > MAX_POINTS) points = MAX_POINTS;
+    
+    // Floor/round and stringify the points, used in description and hover text
+    string current = (string)llFloor(points);
+    string maximum = (string)llRound(MAX_POINTS);
+    
+    // Tweak hover text color based on the points vs max
+    vector color = <1,1,0>; // Amber for ≥20% points
+    if (points >= MAX_POINTS * 0.9) color = <0,1,0>; // Green at ≥90% points
+    else if (points / MAX_POINTS < 0.2) color = <1,0,0>; // Diving into red! <20% points
+    
+    llSetLinkPrimitiveParamsFast(LINK_THIS, [
+        PRIM_HEALTH, points,
+        PRIM_DESC, descPrefix + current + "," + maximum,
+        PRIM_TEXT, "[C2A+LBA]\n" + current + " / " + maximum, color, 1.0
+    ]);
+    
+    if (points == 0) state dead;
+}
+
+
+default
+{
+    state_entry()
+    {
+        points = MAX_POINTS; // Restore to full health
+        
+        llSetLinkPrimitiveParamsFast(LINK_SET, [
+            PRIM_SCRIPTED_SIT_ONLY, TRUE, // Unless a free sit target, don't allow sitting there
+            PRIM_SIT_FLAGS,
+                SIT_FLAG_NO_COLLIDE // Disable avatar collision volume
+            |   SIT_FLAG_NO_DAMAGE // Do not distribute damage to agents 
+        ]);
+        
+        // Listen on LBA channel for backwards compatibility
+        integer lbaChannel = (integer)("0x"+llGetSubString(llMD5String((string)llGetKey(),0),0,3));
+        llListen(lbaChannel, "", "", "");
+        
+        // Generate a unique name for the C2A channel
+        string c2aName;
+        integer n = 8;
+        while (n--) c2aName += llChar(llFloor(llFrand(26)) + 65);
+        llListen(-2453997, c2aName, "", "");
+        descPrefix += c2aName + "],"; // Append it to the description prefix

```

Comment:
If user chooses to respawn via `state default;` there is a flaw here where the string is re-added ontop of the old one. Full descPrefix should be generated here in one go.